packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -78,85 +76,54 @@ const ManagerI = M.interface('manager', {
   wait: M.call(M.bigint()).returns(VowShape),
 });
 
-interface PositionRd {
-  getPoolKey(): PoolKey;
-  getYieldProtocol(): YieldProtocol;
-}
-
-interface PositionPub extends PositionRd {
-  publishStatus(): void;
-}
-
-export interface Position extends PositionPub {
-  recordTransferIn(amount: Amount<'nat'>): Amount<'nat'>;
-  recordTransferOut(amount: Amount<'nat'>): Amount<'nat'>;
-}
-
-export type TransferStatus = {
-  totalIn: Amount<'nat'>;
-  totalOut: Amount<'nat'>;
-  netTransfers: Amount<'nat'>;
+export type AccountInfo = GMPAccountInfo | AgoricAccountInfo | NobleAccountInfo;
+export type GMPAccountInfo = {
+  namespace: 'eip155';
+  chainName: AxelarChain;
+  chainId: CaipChainId;
+  remoteAddress: `0x${string}`;
 };
-
-export const recordTransferIn = (
-  amount: Amount<'nat'>,
-  state: TransferStatus,
-  position: Pick<Position, 'publishStatus'>,
-) => {
-  const { netTransfers, totalIn } = state;
-  assign(state, {
-    netTransfers: add(netTransfers, amount),
-    totalIn: add(totalIn, amount),
-  });
-  position.publishStatus();
-  return state.netTransfers;
+type AgoricAccountInfo = {
+  namespace: 'cosmos';
+  chainName: 'agoric';
+  lca: LocalAccount;
+  reg: TargetRegistration;
 };
-
-export const recordTransferOut = (
-  amount: Amount<'nat'>,
-  state: TransferStatus,
-  position: Pick<Position, 'publishStatus'>,
-) => {
-  const { netTransfers, totalOut } = state;
-  assign(state, {
-    netTransfers: subtract(netTransfers, amount),
-    totalOut: add(totalOut, amount),
-  });
-  position.publishStatus();
-  return state.netTransfers;
+type NobleAccountInfo = {
+  namespace: 'cosmos';
+  chainName: 'noble';
+  ica: NobleAccount;
 };
 
-export type AccountInfoFor = {
-  agoric: { type: 'agoric'; lca: LocalAccount; reg: TargetRegistration };
-  noble: { type: 'noble'; ica: NobleAccount };
+export type AccountInfoFor = Record<AxelarChain, GMPAccountInfo> & {
+  agoric: AgoricAccountInfo;
+  noble: NobleAccountInfo;
 };
 
-export type AccountInfo = AccountInfoFor['agoric'] | AccountInfoFor['noble'];
-// XXX expand scope to GMP
-
-/** keyed by chain such as agoric, noble, base, arbitrum */
-export type ChainAccountKey = 'agoric' | 'noble';
-
 type PortfolioKitState = {
   portfolioId: number;
-  accountsPending: MapStore<ChainAccountKey, VowKit<AccountInfo>>;
-  accounts: MapStore<ChainAccountKey, AccountInfo>;
+  accountsPending: MapStore<SupportedChain, VowKit<AccountInfo>>;
+  accounts: MapStore<SupportedChain, AccountInfo>;
   positions: MapStore<PoolKey, Position>;
   nextFlowId: number;
 };
 
 const accountIdByChain = (accounts: PortfolioKitState['accounts']) => {
```

Comment:
May want to add that this function will throw if any of the account addresses in `cosmos` are not expected `bech32`.  But that's a fine constraint to have if this is for initialisation.